src/NLog/Internal/TargetWithFilterChain.cs

```diff
@@ -92,17 +94,21 @@ internal void PrecalculateStackTraceUsage()
 
             // find all objects which may need stack trace
             // and determine maximum
-            foreach (var item in ObjectGraphScanner.FindReachableObjects<IUsesStackTrace>(this))
+            // only the target can have IUsesStackTrace
+            if (Target != null)
```

Comment:
From what I can see it's definitely needed: The root of the chain is used [here](https://github.com/NLog/NLog/blob/d40035b5f9389e8eef655ccfec56e11a96555b6d/src/NLog/LoggerImpl.cs#L69) to determine which kind of stacktrace is needed, then the chain itself is processed [later](https://github.com/NLog/NLog/blob/d40035b5f9389e8eef655ccfec56e11a96555b6d/src/NLog/LoggerImpl.cs#L97-L103) in a loop.
The PrecalculateStackTraceUsage is also only [called](https://github.com/NLog/NLog/blob/815bf16546ec4d0eec9d758b8d351ab98b036fb2/src/NLog/LogFactory.cs#L783) for the root of the chain, so only the StackTraceUsage of the root is valid.